data_fixtures/migrations/0004_upsert_initial_topic_data.py

```diff
@@ -6,7 +6,20 @@
 
 
 def perform_topic_upsert(apps, schema_editor):
-    """Upsert the topic data from the default location."""
+    """
+    Upsert the topic data from the default location.
+
+    This clears existing topics before upserting, and clears topic channels.
+    After this migration runs, you will need to repopulate these things.
+    """
+
+    LearningResourceTopic = apps.get_model(
+        "learning_resources", "LearningResourceTopic"
+    )
+    ChannelTopicDetail = apps.get_model("channels", "ChannelTopicDetail")
+
+    LearningResourceTopic.objects.all().delete()
+    ChannelTopicDetail.objects.all().delete()
```

Comment:
This still leaves a bunch of orphaned  `Channel` objects with `channel_type="topic"` and a null `ChannelTopicDetail`.  This should handle it, and also delete the related ChannelTopicDetail objects via cascade:

```python
Channel.objects.filter(channel_type="topic").delete()
```

Reply:
Another consequence of deleting all existing topics is that any learning paths (like featured lists for OCW, xPro, etc) will have their manually assigned `topics` field set to an empty list.  Not sure how big a deal this is, as there aren't that many of them, but might be worth checking with @pdpinch.

Reply:
We should run the pipelines for sources that include topic data once this gets merged anyway - would these lists get updated as part of that? (I know there's a `populate_featured_lists` for featured courses, for instance.)